nixos/modules/services/system/swapspace.nix

```
@@ -0,0 +1,92 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.services.swapspace;
+in {
+  ###### interface
+
+  options = {
+
+    services.swapspace = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to create swapfiles dynamically using the SwapSpace manager.
+          Files will be added and removed based on current memory usage.
+        '';
+      };
+
+      path = mkOption {
+        type = types.path;
+        default = "/var/lib/swap";
+        description = ''
+          Location of the swap files. This directory will be restrticted to root.
+        '';
+      };
+
+      cooldown = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = ''
+          Cooldown period between changes.
+          SwapSpace will wait at least this many seconds before an action,
+          like removing a swapfile after adding one.
+        '';
+      };
+
+      minSwapSize = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = ''
+          Minimum size of a swapfile.
+        '';
+      };
+
+      maxSwapSize = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = ''
+          Maximum size of a swapfile.
+        '';
+      };
+
+      extraArgs = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+          Any extra arguments to pass to SwapSpace.
+        '';
+        example = "-P -v -u 5";
+      };
+    };
+  };
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+    systemd.services.swapspace = with pkgs; {
```

```suggestion
    systemd.services.swapspace = {
```

you don't refer to things inside `pkgs` anywhere.